backend/de.metas.externalsystem/src/main/java/de/metas/externalsystem/grssignum/ExportToGRSService.java

```diff
@@ -0,0 +1,105 @@
+ * %%
+ * Copyright (C) 2021 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.externalsystem.grssignum;
+
+import de.metas.audit.data.repository.DataExportAuditLogRepository;
+import de.metas.audit.data.repository.DataExportAuditRepository;
+import de.metas.bpartner.BPartnerId;
+import de.metas.common.externalsystem.ExternalSystemConstants;
+import de.metas.externalsystem.ExternalSystemConfigRepo;
+import de.metas.externalsystem.ExternalSystemConfigService;
+import de.metas.externalsystem.ExternalSystemType;
+import de.metas.externalsystem.IExternalSystemChildConfig;
+import de.metas.externalsystem.export.bpartner.ExportToExternalSystemService;
+import de.metas.externalsystem.rabbitmq.ExternalSystemMessageSender;
+import de.metas.util.Check;
+import lombok.NonNull;
+import org.adempiere.exceptions.AdempiereException;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class ExportToGRSService extends ExportToExternalSystemService
+{
+	private static final String EXTERNAL_SYSTEM_COMMAND_EXPORT_BPARTNER = "exportBPartner";
+
+	public ExportToGRSService(
+			@NonNull final ExternalSystemConfigRepo externalSystemConfigRepo,
+			@NonNull final DataExportAuditRepository dataExportAuditRepository,
+			@NonNull final DataExportAuditLogRepository dataExportAuditLogRepository,
+			@NonNull final ExternalSystemMessageSender externalSystemMessageSender,
+			@NonNull final ExternalSystemConfigService externalSystemConfigService)
+	{
+		super(externalSystemConfigRepo, externalSystemMessageSender, dataExportAuditLogRepository, dataExportAuditRepository, externalSystemConfigService);
+	}
+
+	@Override
+	@NonNull
+	protected Map<String, String> buildParameters(
+			@NonNull final IExternalSystemChildConfig childConfig,
+			@NonNull final BPartnerId bPartnerId)
+	{
+		final ExternalSystemGRSSignumConfig grsSignumConfig = ExternalSystemGRSSignumConfig.cast(childConfig);
+
+		final Map<String, String> parameters = new HashMap<>();
+
+		final String authToken = grsSignumConfig.getAuthToken();
+
+		if (Check.isBlank(authToken))
+		{
+			throw new AdempiereException("Missing authToken for ExternalSystemGRSSignumConfig")
+					.appendParametersToMessage()
+					.setParameter("ExternalSystemGRSSignumConfigId", grsSignumConfig.getId());
+		}
```

Comment:
```suggestion
```

we don't know that the Auth-Token is actually a thing, that's why i spec'ed it as optional